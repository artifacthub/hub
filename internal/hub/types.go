package hub

type userIDKey struct{}

// UserIDKey represents the key used for the userID value inside a context.
var UserIDKey = userIDKey{}

// ChartRepository represents a Helm chart repository.
type ChartRepository struct {
	ChartRepositoryID string `json:"chart_repository_id"`
	Name              string `json:"name"`
	DisplayName       string `json:"display_name"`
	URL               string `json:"url"`
	UserID            string `json:"user_id"`
}

// Link represents a url associated with a package.
type Link struct {
	Name string `json:"name"`
	URL  string `json:"url"`
}

// Maintainer represents a package's maintainer.
type Maintainer struct {
	MaintainerID string `json:"maintainer_id"`
	Name         string `json:"name"`
	Email        string `json:"email"`
}

// PackageKind represents the kind of a given package.
type PackageKind int64

const (
	// Chart represents a Helm chart.
	Chart PackageKind = 0

	// Falco represents a set of Falco rules.
	Falco PackageKind = 1

	// OPA represents a set of OPA policies.
	OPA PackageKind = 2
)

// Package represents a Kubernetes package.
type Package struct {
	PackageID         string                 `json:"package_id"`
	Kind              PackageKind            `json:"kind"`
	Name              string                 `json:"name"`
	DisplayName       string                 `json:"display_name"`
	Description       string                 `json:"description"`
	HomeURL           string                 `json:"home_url"`
	LogoURL           string                 `json:"logo_url"`
	LogoImageID       string                 `json:"logo_image_id"`
	Keywords          []string               `json:"keywords"`
	Deprecated        bool                   `json:"deprecated"`
	Readme            string                 `json:"readme"`
	Links             []*Link                `json:"links"`
	Version           string                 `json:"version"`
	AvailableVersions []string               `json:"available_versions"`
	AppVersion        string                 `json:"app_version"`
	Digest            string                 `json:"digest"`
	Data              map[string]interface{} `json:"data"`
	Maintainers       []*Maintainer          `json:"maintainers"`
	ChartRepository   *ChartRepository       `json:"chart_repository"`
}

// SearchPackageInput represents the query input when searching for packages.
type SearchPackageInput struct {
	Limit             int           `json:"limit,omitempty"`
	Offset            int           `json:"offset,omitempty"`
	Facets            bool          `json:"facets"`
	Text              string        `json:"text"`
	PackageKinds      []PackageKind `json:"package_kinds,omitempty"`
	ChartRepositories []string      `json:"chart_repositories,omitempty"`
	Deprecated        bool          `json:"deprecated"`
}

// GetPackageInput represents the input used to get a specific package.
type GetPackageInput struct {
	ChartRepositoryName string `json:"chart_repository_name"`
	PackageName         string `json:"package_name"`
	Version             string `json:"version"`
}

// User represents a Hub user.
type User struct {
	UserID        string `json:"user_id"`
	Alias         string `json:"alias"`
	FirstName     string `json:"first_name"`
	LastName      string `json:"last_name"`
	Email         string `json:"email"`
	EmailVerified bool   `json:"email_verified"`
	Password      string `json:"password"`
}

// CheckCredentialsOutput represents the output returned by the
// CheckCredentials method.
type CheckCredentialsOutput struct {
	Valid  bool   `json:"valid"`
	UserID string `json:"user_id"`
}

// Session represents some information about a user session.
type Session struct {
	SessionID string `json:"session_id"`
	UserID    string `json:"user_id"`
	IP        string `json:"ip"`
	UserAgent string `json:"user_agent"`
}

// CheckSessionOutput represents the output returned by the CheckSession method.
type CheckSessionOutput struct {
	Valid  bool   `json:"valid"`
	UserID string `json:"user_id"`
}

// Organization represents an entity with one or more users associated that can
// own packages and other entities like chart repositories.
type Organization struct {
	OrganizationID string `json:"organization_id"`
	Name           string `json:"name"`
	DisplayName    string `json:"display_name"`
	Description    string `json:"description"`
	HomeURL        string `json:"home_url"`
	LogoURL        string `json:"logo_url"`
	LogoImageID    string `json:"logo_image_id"`
}
