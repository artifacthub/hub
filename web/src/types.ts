import { JSONSchema } from '@apidevtools/json-schema-ref-parser';

export enum RepositoryKind {
  Helm = 0,
  Falco,
  OPA,
  OLM,
  TBAction,
  Krew,
  HelmPlugin,
}

export interface Repository {
  repositoryId?: string;
  name: string;
  displayName?: string | null;
  url: string;
  branch?: string | null;
  organizationName?: string | null;
  organizationDisplayName?: string | null;
  userAlias?: string | null;
  kind: RepositoryKind;
  lastTrackingTs?: number | null;
  lastTrackingErrors?: string | null;
  verifiedPublisher?: boolean;
  official?: boolean;
  private?: boolean;
  authUser?: string | null;
  authPass?: string | null;
  disabled?: boolean;
  scannerDisabled?: boolean;
}

export interface Maintainer {
  name?: string;
  email: string;
}

export interface PackageLink {
  name: string;
  url: string;
}

export interface Channel {
  name: string;
  version: string;
}

export interface Package {
  packageId: string;
  name: string;
  displayName: string | null;
  normalizedName: string;
  description: string;
  logoImageId?: string;
  appVersion?: string;
  repository: Repository;
  readme?: string | null;
  install?: string | null;
  data?: PackageData | null;
  availableVersions?: Version[];
  version?: string;
  homeUrl?: string | null;
  keywords?: string[];
  maintainers?: Maintainer[];
  deprecated: boolean | null;
  isOperator?: boolean | null;
  signed: boolean | null;
  links?: PackageLink[];
  stars?: number | null;
  eventKinds?: EventKind[];
  license?: string | null;
  createdAt: number;
  defaultChannel?: string | null;
  channels?: Channel[] | null;
  provider?: string | null;
  containersImages?: ContainerImage[] | null;
  capabilities?: string | null;
  crds?: { [key: string]: any } | null;
  crdsExamples?: CustomResourcesDefinitionExample[] | null;
  securityReportSummary?: SecurityReportSummary | null;
  securityReportCreatedAt?: number;
  hasValuesSchema?: boolean;
  hasChangelog?: boolean;
  contentUrl?: string;
  containsSecurityUpdates?: boolean;
  prerelease?: boolean;
}

export interface ContainerImage {
  image: string;
  name?: string;
}

export interface Version {
  version: string;
  containsSecurityUpdates: boolean;
  prerelease: boolean;
  createdAt: number;
}

export interface OLMExtraData {
  customResourcesExamples?: string;
}

export interface CustomResourcesDefinitionExample {
  kind: string;
  [key: string]: any;
}

export interface CustomResourcesDefinition {
  kind: string;
  name: string;
  version: string;
  description: string;
  displayName?: string;
  example?: CustomResourcesDefinitionExample;
}

export interface Dependency {
  name: string;
  version: string;
  repository?: string;
}

export interface PackageData {
  policies?: OPAPolicies;
  rules?: { Raw: string }[] | FalcoRules;
  dependencies?: Dependency[];
  customResourcesDefinitionsExamples?: string;
  customResourcesDefinitions?: CustomResourcesDefinition[];
  isGlobalOperator?: boolean;
}

export interface OPAPolicies {
  [key: string]: string;
}

export interface FalcoRules {
  [key: string]: string;
}

export interface SearchFiltersURL {
  tsQueryWeb?: string;
  tsQuery?: string[];
  filters: {
    [key: string]: string[];
  };
  pageNumber: number;
  deprecated?: boolean | null;
  operators?: boolean | null;
  verifiedPublisher?: boolean | null;
  official?: boolean | null;
}

export interface SearchQuery {
  tsQueryWeb?: string;
  tsQuery?: string[];
  filters: {
    [key: string]: string[];
  };
  deprecated?: boolean | null;
  operators?: boolean | null;
  verifiedPublisher?: boolean | null;
  official?: boolean | null;
  limit: number;
  offset: number;
  total?: number;
}

export interface SearchResults {
  data: {
    packages: Package[] | null;
    facets: Facets[] | null;
  };
  metadata: {
    limit: number;
    offset: number;
    total: number;
  };
}

export interface Stats {
  packages: number;
  releases: number;
}

export interface Facets {
  filterKey: string;
  title: string;
  options: FacetOption[];
}

export interface FacetOption {
  id: string | number;
  name: string;
  total?: number;
}

export interface Option extends FacetOption {
  filterKey: string;
  icon?: JSX.Element;
}

export interface UserLogin {
  email: string;
  password: string;
}

export interface UserFullName {
  alias: string;
  firstName?: string;
  lastName?: string;
  profileImageId?: null | string;
}

export interface Profile extends UserFullName {
  email: string;
}

export interface User extends UserLogin {
  alias: string;
  firstName?: string;
  lastName?: string;
}

export interface Member {
  alias: string;
  firstName?: string;
  lastName?: string;
  confirmed?: boolean;
}

export interface UserAuth {
  alias?: string;
  status: boolean;
}

export interface CheckAvailabilityProps {
  resourceKind: ResourceKind;
  value: string;
}

export enum ResourceKind {
  userAlias = 'userAlias',
  repositoryName = 'repositoryName',
  repositoryURL = 'repositoryURL',
  organizationName = 'organizationName',
}

export interface Organization {
  name: string;
  displayName?: string;
  homeUrl?: string;
  logoImageId?: string;
  description?: string;
  membersCount?: number | null;
  confirmed?: boolean | null;
}

export interface RefInputField {
  checkIsValid: () => Promise<boolean>;
  reset: () => void;
  getValue: () => string;
  checkValidity: () => boolean;
  updateValue: (value: string) => void;
}

export interface RefInputTypeaheadField {
  reset: () => void;
  getValue: () => string;
  updateValue: (value: string) => void;
}

export interface RefActionBtn {
  reRender: () => void;
}

export interface Alert {
  type: 'success' | 'danger' | 'warning';
  message: string;
  dismissOn?: number;
  autoClose?: boolean;
}

export interface LogoImage {
  imageId: string;
}

export interface AvailabilityInfo {
  isAvailable: boolean;
  resourceKind: ResourceKind;
  excluded: string[];
}

export interface Prefs {
  theme: ThemePrefs;
  controlPanel: {
    selectedOrg?: string;
  };
  search: {
    limit: number;
  };
}

export interface ThemePrefs {
  configured: string;
  efective?: string;
  automatic: boolean;
}

export interface PackageStars {
  starredByUser?: boolean | null;
  stars: number | null;
}

export enum EventKind {
  NewPackageRelease = 0,
  SecurityAlert,
  RepositoryTrackingErrors,
}

export interface Subscription {
  eventKind: EventKind;
}

export interface TestWebhook {
  url: string;
  contentType?: string | null;
  template?: string | null;
  eventKinds: EventKind[];
}

export interface Webhook extends TestWebhook {
  webhookId?: string;
  name: string;
  description?: string;
  secret?: string;
  active: boolean;
  packages: Package[];
  lastNotifications?: null | WebhookNotification[];
}

export interface WebhookNotification {
  notificationId: string;
  createdAt: number;
  processed: boolean;
  processedAt: number;
  error?: null | string;
}

export enum PayloadKind {
  default = 0,
  custom,
}

export interface Section {
  name: string;
  displayName: string;
  shortName?: string;
  disabled: boolean;
  onlyDesktop?: boolean;
  icon?: JSX.Element;
  subsections?: Section[];
}

export interface NavSection {
  user: Section[];
  org: Section[];
}

export interface APIKey {
  apiKeyId?: string;
  name: string;
  createdAt?: number;
}

export interface APIKeyCode {
  key: string;
}

export interface OptionWithIcon {
  value: string;
  label: string;
  icon: JSX.Element;
}

export interface TsQuery {
  name: string;
  label: string;
  value: string;
}
export interface Error {
  kind: ErrorKind;
  message?: string;
}

export enum ErrorKind {
  Other,
  NotFound,
  Unauthorized,
  Forbidden,
}

export interface OptOutItem {
  optOutId: string;
  repository: Repository;
  eventKind: EventKind;
}

export enum AuthorizerAction {
  AddOrganizationMember = 'addOrganizationMember',
  AddOrganizationRepository = 'addOrganizationRepository',
  DeleteOrganization = 'deleteOrganization',
  DeleteOrganizationMember = 'deleteOrganizationMember',
  DeleteOrganizationRepository = 'deleteOrganizationRepository',
  GetAuthorizationPolicy = 'getAuthorizationPolicy',
  TransferOrganizationRepository = 'transferOrganizationRepository',
  UpdateAuthorizationPolicy = 'updateAuthorizationPolicy',
  UpdateOrganization = 'updateOrganization',
  UpdateOrganizationRepository = 'updateOrganizationRepository',
}

export interface AuthorizerInput {
  action: AuthorizerAction;
  organizationName?: string;
  user: string;
  onCompletion?: () => void;
}

export interface RegoPlaygroundPolicy {
  rego_modules: {
    'policy.rego': string;
  };
  input: {
    [key: string]: string | string[];
  };
  data: {
    [key: string]: any;
  };
}

export interface RegoPlaygroundResult {
  result: string;
}

export interface OrganizationPolicy {
  authorizationEnabled: boolean;
  predefinedPolicy?: string | null;
  customPolicy?: string | null;
  policyData?: string | null;
}

export interface AuthorizationPolicy {
  name: string;
  label: string;
  policy: string;
  data: {
    [key: string]: any;
  };
}

export interface SearchTipItem {
  content: JSX.Element | string;
  example: string;
}

export type SecurityReportSummary = {
  [key in VulnerabilitySeverity]?: number;
};

export interface SecurityReport {
  [key: string]: SecurityTargetReport[];
}

export interface SecurityTargetReport {
  Target: string;
  Type: string;
  Vulnerabilities: Vulnerability[] | null;
}

export interface Vulnerability {
  [key: string]: any;
  Severity: VulnerabilitySeverity;
}

export enum VulnerabilitySeverity {
  Critical = 'critical',
  High = 'high',
  Medium = 'medium',
  Low = 'low',
  UnKnown = 'unknown',
}

export interface SeverityRating {
  color: string;
  level: 'A' | 'B' | 'C' | 'D' | 'F' | '-';
}

export type SeverityRatingList = {
  [key in VulnerabilitySeverity | 'default']?: SeverityRating;
};

export interface CVSSVectorMetric {
  value: string;
  label: string;
  options: CVSSVectorOpt[];
}

export interface CVSSVectorOpt {
  value: string;
  label: string;
  level: number;
}

export interface ChangeLog {
  version: string;
  createdAt: number;
  prerelease: boolean;
  containsSecurityUpdates: boolean;
  changes?: string[] | null;
}

export interface ActiveJSONSchemaValue {
  active: number;
  combinationType: string | null;
  options: JSONSchema[];
  error?: boolean;
}
