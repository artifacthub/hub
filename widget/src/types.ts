export interface PackageSummary {
  packageId: string;
  name: string;
  displayName?: string;
  normalizedName: string;
  description: string;
  logoImageId?: string;
  repository: Repository;
  version: string;
  stars?: number | null;
  ts: number;
  official?: boolean;
}

export interface Repository {
  kind: RepositoryKind;
  name: string;
  verifiedPublisher?: boolean;
  official?: boolean;
  organizationName?: string;
  organizationDisplayName?: string;
  userAlias?: string;
}

export enum RepositoryKind {
  Helm = 0,
  Falco,
  OPA,
  OLM,
  TBAction,
  Krew,
  HelmPlugin,
  TektonTask,
  KedaScaler,
  CoreDNS,
  Keptn,
  TektonPipeline,
  Container,
  Kubewarden,
}

export interface SearchResults {
  packages: PackageSummary[] | null;
  facets: null;
}
