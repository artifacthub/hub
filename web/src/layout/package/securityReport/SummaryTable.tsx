import React from 'react';

import { SecurityReport, SecurityTargetReport, Vulnerability, VulnerabilitySeverity } from '../../../types';
import { SEVERITY_ORDER } from '../../../utils/data';
import formatSecurityReport from '../../../utils/formatSecurityReport';
import sumObjectValues from '../../../utils/sumObjectValues';
import SecurityRating from '../../common/SecutityRating';
import styles from './SummaryTable.module.css';

interface Props {
  report: SecurityReport;
}

const SummaryTable = (props: Props) => {
  return (
    <div className="my-4 d-none d-lg-block">
      <table className={`table table-bordered table-md ${styles.table}`}>
        <thead>
          <tr className={`table-primary ${styles.tableTitle}`}>
            <th className={styles.image} scope="col">
              Image
            </th>
            <th scope="col" className="text-center">
              Rating
            </th>
            {SEVERITY_ORDER.map((severity: VulnerabilitySeverity) => (
              <th key={`col_${severity}`} className="col text-center text-capitalize">
                {severity}
              </th>
            ))}
            <th scope="col" className="text-center">
              Total
            </th>
          </tr>
        </thead>
        <tbody>
          {Object.keys(props.report).map((image: string, index: number) => {
            let vulnerabilitiesList: Vulnerability[] = [];
            props.report[image].forEach((targetReport: SecurityTargetReport) => {
              if (targetReport.Vulnerabilities) {
                vulnerabilitiesList = [...vulnerabilitiesList, ...targetReport.Vulnerabilities];
              }
            });
            const { summary } = formatSecurityReport(vulnerabilitiesList);
            const total = sumObjectValues(summary);

            return (
              <tr key={`tr_${image}`}>
                <td>
                  <div className={styles.imageNameWrapper}>
                    <div className="text-truncate">{image}</div>
                  </div>
                </td>
                <td className="text-center">
                  <SecurityRating summary={summary} onlyBadge />
                </td>
                {SEVERITY_ORDER.map((severity: VulnerabilitySeverity) => (
                  <td key={`col_${severity}_${image}`} className="text-center">
                    {summary[severity] || 0}
                  </td>
                ))}
                <td className="text-center font-weight-bold">{total}</td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

export default SummaryTable;
