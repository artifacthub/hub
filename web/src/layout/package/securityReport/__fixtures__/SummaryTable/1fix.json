{
  "report": {
    "rook/ceph:v1.1.1 (centos 7.7.1908)": {
      "summary": { "critical": 1, "high": 0, "medium": 3, "low": 1, "unknown": 0 },
      "total": 5
    },
    "usr/share/ceph/mgr/dashboard/frontend/package-lock.json": {
      "summary": { "critical": 0, "high": 0, "medium": 1, "low": 1, "unknown": 0 },
      "total": 5
    }
  },
  "summary": { "critical": 1, "high": 0, "medium": 4, "low": 2, "unknown": 0 },
  "total": 7
}
