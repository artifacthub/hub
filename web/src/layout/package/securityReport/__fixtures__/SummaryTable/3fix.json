{
  "report": {
    "artifacthub/hub:v0.7.0": {
      "summary": { "critical": 1, "high": 0, "medium": 3, "low": 1, "unknown": 0 },
      "total": 5
    },
    "artifacthub/scanner:v0.7.0": {
      "summary": { "critical": 0, "high": 0, "medium": 0, "low": 0, "unknown": 0 },
      "total": 0
    },
    "artifacthub/tracker:v0.7.0": {
      "summary": { "critical": 0, "high": 0, "medium": 0, "low": 0, "unknown": 0 },
      "total": 0
    },
    "artifacthub/db-migrator:v0.7.0": {
      "summary": { "critical": 0, "high": 0, "medium": 0, "low": 0, "unknown": 0 },
      "total": 0
    }
  },
  "summary": { "critical": 1, "high": 0, "medium": 3, "low": 1, "unknown": 0 },
  "total": 7
}
