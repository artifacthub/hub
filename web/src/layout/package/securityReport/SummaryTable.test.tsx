import { render } from '@testing-library/react';
import React from 'react';

import { SecurityReport } from '../../../types';
import SummaryTable from './SummaryTable';

const getMockSecurityReport = (fixtureId: string): SecurityReport => {
  return require(`./__fixtures__/SummaryTable/${fixtureId}.json`) as SecurityReport;
};

describe('SummaryTable', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const result = render(<SummaryTable report={getMockSecurityReport('1')} />);
    expect(result.asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders component', () => {
      const { getByText } = render(<SummaryTable report={getMockSecurityReport('2')} />);
      expect(getByText('Image')).toBeInTheDocument();
      expect(getByText('Rating')).toBeInTheDocument();
      expect(getByText('critical')).toBeInTheDocument();
      expect(getByText('high')).toBeInTheDocument();
      expect(getByText('medium')).toBeInTheDocument();
      expect(getByText('low')).toBeInTheDocument();
      expect(getByText('unknown')).toBeInTheDocument();
      expect(getByText('Total')).toBeInTheDocument();
      expect(getByText('rook/ceph:v1.1.1')).toBeInTheDocument();
      expect(getByText('F')).toBeInTheDocument();
      expect(getByText('4')).toBeInTheDocument();
      expect(getByText('22')).toBeInTheDocument();
      expect(getByText('461')).toBeInTheDocument();
      expect(getByText('455')).toBeInTheDocument();
      expect(getByText('0')).toBeInTheDocument();
      expect(getByText('942')).toBeInTheDocument();
    });

    it('renders table with more than one image', () => {
      const { getByText, getAllByText } = render(<SummaryTable report={getMockSecurityReport('3')} />);
      expect(getByText('Image')).toBeInTheDocument();
      expect(getByText('Rating')).toBeInTheDocument();
      expect(getByText('critical')).toBeInTheDocument();
      expect(getByText('high')).toBeInTheDocument();
      expect(getByText('medium')).toBeInTheDocument();
      expect(getByText('low')).toBeInTheDocument();
      expect(getByText('unknown')).toBeInTheDocument();
      expect(getByText('Total')).toBeInTheDocument();
      expect(getByText('artifacthub/hub:v0.7.0')).toBeInTheDocument();
      expect(getByText('artifacthub/scanner:v0.7.0')).toBeInTheDocument();
      expect(getByText('artifacthub/tracker:v0.7.0')).toBeInTheDocument();
      expect(getByText('artifacthub/db-migrator:v0.7.0')).toBeInTheDocument();
      expect(getByText('F')).toBeInTheDocument();
      expect(getAllByText('A')).toHaveLength(3);
      expect(getAllByText('1')).toHaveLength(2);
      expect(getAllByText('0')).toHaveLength(20);
      expect(getByText('7')).toBeInTheDocument();
    });
  });
});
