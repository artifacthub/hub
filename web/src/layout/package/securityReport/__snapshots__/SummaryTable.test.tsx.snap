// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryTable creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-4 d-none d-lg-block"
  >
    <table
      class="table table-bordered table-md table"
    >
      <thead>
        <tr
          class="table-primary tableTitle"
        >
          <th
            class="image"
            scope="col"
          >
            Image
          </th>
          <th
            class="text-center"
            scope="col"
          >
            Rating
          </th>
          <th
            class="col text-center text-capitalize"
          >
            critical
          </th>
          <th
            class="col text-center text-capitalize"
          >
            high
          </th>
          <th
            class="col text-center text-capitalize"
          >
            medium
          </th>
          <th
            class="col text-center text-capitalize"
          >
            low
          </th>
          <th
            class="col text-center text-capitalize"
          >
            unknown
          </th>
          <th
            class="text-center"
            scope="col"
          >
            Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div
              class="imageNameWrapper"
            >
              <div
                class="text-truncate"
              >
                rook/ceph:v1.1.1
              </div>
            </div>
          </td>
          <td
            class="text-center"
          >
            <div
              class="position-relative undefined"
            >
              <div
                data-testid="elementWithTooltip"
              >
                <small>
                  <div
                    class="badge badge-pill text-light font-weight-bold badge undefined"
                    style="background-color: rgb(150, 0, 3);"
                  >
                    F
                  </div>
                </small>
              </div>
            </div>
          </td>
          <td
            class="text-center"
          >
            4
          </td>
          <td
            class="text-center"
          >
            22
          </td>
          <td
            class="text-center"
          >
            461
          </td>
          <td
            class="text-center"
          >
            455
          </td>
          <td
            class="text-center"
          >
            0
          </td>
          <td
            class="text-center font-weight-bold"
          >
            942
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
